src/OVAL/oval_agent.c

```diff
@@ -169,13 +170,14 @@ int oval_agent_eval_definition(oval_agent_session_t *ag_sess, const char *id)
 		title = oval_definition_get_title(oval_def);
 	}
 	dI("Evaluating definition '%s': %s.", id, title);
+	rsystem = _oval_agent_get_first_result_system(ag_sess);
+	oval_probe_session_set_result_system(ag_sess->psess, rsystem);
```

Comment:
We really should think about handling this differently.

Please draw a diagram of relations between `agent_session`, `results_model`, `result_system`, `probe_session`, and `syschar_model`. You will see the madness. :-)

From the diagram. I think we should never allow to `probe_session_set_result_system`. It make sense only for `result_system` assigned with the same `syschar_model`.

What about creating new oval_probe_session_new which would take `result_system` and deprecating the other one?